File: frontend/src/api/activity.ts
import request from '../utils/request'

export interface Checkpoint {
  id: number
  activity_id: number
  name: string
  sequence: number
  lat: number
  lng: number
  clue?: string
  task_type: string
  task_content?: string
  radius_meters: number
  qr_code: string
}

export interface Activity {
  id: number
  title: string
  description?: string
  difficulty: string
  duration_minutes: number
  equipment_requirement?: string
  start_time: string
  end_time: string
  status: string
  creator_id: number
  start_lat: number
  start_lng: number
  end_lat: number
  end_lng: number
  address: string
  max_teams: number
  checkpoint_count: number
  team_count: number
  checkpoints?: Checkpoint[]
  created_at: string
}

export interface ActivityPayload {
  title: string
  description?: string
  difficulty: string
  duration_minutes: number
  equipment_requirement?: string
  start_time: string
  end_time: string
  start_lat: number
  start_lng: number
  end_lat: number
  end_lng: number
  address: string
  max_teams: number
}

export function listActivities(params: { page?: number; page_size?: number; status?: string; difficulty?: string; keyword?: string }) {
  return request.get<{ list: Activity[]; total: number; page: number; page_size: number }>('/activities', { params })
}

export function getActivity(id: number) {
  return request.get<Activity>(`/activities/${id}`)
}

export function createActivity(data: ActivityPayload) {
  return request.post('/activities', data)
}

export function updateActivity(id: number, data: ActivityPayload) {
  return request.put(`/activities/${id}`, data)
}

export function transitionActivity(id: number, status: string) {
  return request.post(`/activities/${id}/transition`, { status })
}

export function deleteActivity(id: number) {
  return request.delete(`/activities/${id}`)
}

File: backend/internal/model/activity.go
package model

import "time"

// Activity 城市定向越野活动（线路）实体。
type Activity struct {
	ID                 int64     `gorm:"primaryKey" json:"id"`
	Title              string    `gorm:"size:120;not null" json:"title"`
	Description        string    `gorm:"type:text" json:"description"`
	Difficulty         string    `gorm:"size:20;default:adult;not null" json:"difficulty"`
	DurationMinutes    int       `gorm:"default:120;not null" json:"duration_minutes"`
	EquipmentRequirement string  `gorm:"type:text" json:"equipment_requirement"`
	StartTime          time.Time `json:"start_time"`
	EndTime            time.Time `json:"end_time"`
	Status             string    `gorm:"size:20;default:draft;not null;index" json:"status"`
	CreatorID          int64     `gorm:"index;not null" json:"creator_id"`
	StartLat           float64   `json:"start_lat"`
	StartLng           float64   `json:"start_lng"`
	EndLat             float64   `json:"end_lat"`
	EndLng             float64   `json:"end_lng"`
	Address            string    `gorm:"size:255" json:"address"`
	MaxTeams           int       `gorm:"default:50;not null" json:"max_teams"`
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`

	Checkpoints []Checkpoint `gorm:"foreignKey:ActivityID" json:"checkpoints,omitempty"`
}
